In [8]:
#importing dataset

import pandas as pd
import numpy as np

df = pd.read_csv(r'C:\Users\Preetham.Reddy\Retail-Sales-Data\data\raw\orders.csv')
df

,order_id,order_date,brand,model_name,category,gender,size,color,base_price_usd,discount_percent,final_price_usd,units_sold,revenue_usd,payment_method,sales_channel,country,customer_income_level,customer_rating
0,ORD100000,2021-01-30,ASICS,Model-370,Running,Unisex,8,Black,162,15,137.70,1,137.70,Card,Retail Store,Germany,Low,4.6
1,ORD100001,2026-10-05,Reebok,Model-314,Lifestyle,Men,8,Grey,80,5,76.00,3,228.00,Card,Online,USA,Low,3.9
2,ORD100002,2023-11-12,ASICS,Model-763,Lifestyle,Men,8,Black,176,15,149.60,4,598.40,Cash,Retail Store,India,Medium,3.0
3,ORD100003,2026-08-29,Reebok,Model-905,Basketball,Women,7,White,61,15,51.85,2,103.70,Card,Retail Store,India,High,3.4
4,ORD100004,2019-11-09,Nike,Model-413,Training,Men,11,Black,80,0,80.00,4,320.00,Cash,Online,USA,Medium,3.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
29995,ORD129995,2018-05-28,Puma,Model-892,Gym,Women,6,Blue,140,10,126.00,1,126.00,Card,Online,UAE,High,4.3
29996,ORD129996,2018-04-28,Adidas,Model-982,Running,Women,9,Grey,113,15,96.05,1,96.05,Card,Online,UK,High,3.5
29997,ORD129997,2026-02-17,ASICS,Model-463,Basketball,Men,11,Blue,130,30,91.00,1,91.00,Wallet,Online,India,Medium,4.4
29998,ORD129998,2024-02-21,New Balance,Model-984,Running,Unisex,11,White,186,10,167.40,2,334.80,Wallet,Retail Store,USA,High,3.9


In [ ]:
#section1 : data quality and validation

In [9]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 18 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   order_id               30000 non-null  object 
 1   order_date             30000 non-null  object 
 2   brand                  30000 non-null  object 
 3   model_name             30000 non-null  object 
 4   category               30000 non-null  object 
 5   gender                 30000 non-null  object 
 6   size                   30000 non-null  int64  
 7   color                  30000 non-null  object 
 8   base_price_usd         30000 non-null  int64  
 9   discount_percent       30000 non-null  int64  
 10  final_price_usd        30000 non-null  float64
 11  units_sold             30000 non-null  int64  
 12  revenue_usd            30000 non-null  float64
 13  payment_method         30000 non-null  object 
 14  sales_channel          30000 non-null  object 
 15  co

In [10]:
#converting order_date column  to date format

df['order_date_parsed'] = pd.to_datetime(df['order_date'])

In [11]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 30000 entries, 0 to 29999
Data columns (total 19 columns):
 #   Column                 Non-Null Count  Dtype         
---  ------                 --------------  -----         
 0   order_id               30000 non-null  object        
 1   order_date             30000 non-null  object        
 2   brand                  30000 non-null  object        
 3   model_name             30000 non-null  object        
 4   category               30000 non-null  object        
 5   gender                 30000 non-null  object        
 6   size                   30000 non-null  int64         
 7   color                  30000 non-null  object        
 8   base_price_usd         30000 non-null  int64         
 9   discount_percent       30000 non-null  int64         
 10  final_price_usd        30000 non-null  float64       
 11  units_sold             30000 non-null  int64         
 12  revenue_usd            30000 non-null  float64       
 13  p

In [16]:
df[["order_date","order_date_parsed"]]

,order_date,order_date_parsed
0,2021-01-30,2021-01-30
1,2026-10-05,2026-10-05
2,2023-11-12,2023-11-12
3,2026-08-29,2026-08-29
4,2019-11-09,2019-11-09
...,...,...
29995,2018-05-28,2018-05-28
29996,2018-04-28,2018-04-28
29997,2026-02-17,2026-02-17
29998,2024-02-21,2024-02-21


In [18]:
invalid_date_count = df["order_date_parsed"].isna().sum()
invalid_date_count

0

In [26]:
today = pd.Timestamp.today().normalize()
#today
future_date_count = (df["order_date_parsed"]>today).sum()
future_date_count

3079

In [27]:
old_date = pd.Timestamp("2018-01-01")
old_orders = (df["order_date_parsed"]<old_date).sum()
old_orders

0

### findings

-invalid dates : 0 rows
-future orders : 3079 orders
-orders older than 2018 : 0 orders

### price and revenue consistency
We are trying to confirm if base_price, discounts and final_prices look accurate and also units_sold and revenue_USD.

In [43]:
#Validate whether final_price_usd is correctly derived from base price and discount.

df["final_price_validation"] = (df["base_price_usd"]*(1-(df["discount_percent"]/100)))*df['units_sold']
wrong_price = ((df['revenue_usd']-df['final_price_validation'])>0.1).sum()
wrong_price


0

### findings
number of inconsistent rows : 0

In [ ]:
#next steps